### ISSUE TITLE
Clarify how `Plane3d` is oriented in space

### ISSUE THREAD

**Issue by mweatherley** [bevyengine/bevy#13019]
## How can Bevy's documentation be improved?

We have a primitive, `Plane3d`, which really describes a rectangular section of an infinite plane. At the level of the type itself, it is described by its `normal` and a `half-size`:
```rust
pub struct Plane3d {
    /// The normal of the plane. The plane will be placed perpendicular to this direction
    pub normal: Dir3,
    /// Half of the width and height of the plane
    pub half_size: Vec2,
}
```
The documentation doesn't actually say how the rectangle is rotated along the axis of the normal, although this is implicit, for example, in its implementation of `Meshable`.

Notably, this area is also a little bit thorny, since it's [impossible](https://en.wikipedia.org/wiki/Hairy_ball_theorem) to have a consistent (continuous) orientation for this rectangle over the entire sphere of possible normals. Right now, the thing implied by `Meshable` is underdetermined at `normal == Vec::NEG_Y`.

**Comment by NthTensor** [comment]
Might be better to use a private pair of orthogonal basis vectors. Would downgrade normal to a function, so would be a breaking change.

**Comment by mweatherley** [comment]
> Might be better to use a private pair of orthonormal vectors. Would downgrade normal to a function, so would be a breaking change.

I like this idea.

**Comment by rlidwka** [comment]
@mweatherley, I have a question though. If Plane3d represents a "rectangular section of an infinite plane", then what does Plane2d represent?

```rs
pub struct Plane3d {
    /// The normal of the plane. The plane will be placed perpendicular to this direction
    pub normal: Dir3,
    /// Half of the width and height of the plane
    pub half_size: Vec2,
}

pub struct Plane2d {
    /// The normal of the plane. The plane will be placed perpendicular to this direction
    pub normal: Dir2,
}                                                                                                                       
```

I believe these two things should be pushed back to the drawing board.

**Comment by NthTensor** [comment]
I think plane2d is actually an finite line iirc. Yeah, the situation is not great.

**Comment by Jondolf** [comment]
Honestly, I'm not sure we even need `Plane2d` since we already have `Line2d`. The only difference is just that they're perpendicular to one another.

**Comment by NthTensor** [comment]
Good point. The only thing `Line2d` is missing is a right-handed `normal() -> Dir2` method.
